config/initializers/secure_headers.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# rubocop:disable Lint/PercentStringArray
+SecureHeaders::Configuration.default do |config|
+  config.x_frame_options = "DENY"
+  config.x_content_type_options = "nosniff"
+  config.x_xss_protection = "1; mode=block"
+  config.x_download_options = "noopen"
+  config.x_permitted_cross_domain_policies = "none"
+  config.referrer_policy = %w[origin-when-cross-origin strict-origin-when-cross-origin]
+
+  google_analytics = %w[www.google-analytics.com ssl.google-analytics.com *.googletagmanager.com tagmanager.google.com *.googleusercontent.com *.gstatic.com]
+
+  config.csp = {
+    default_src: %w['none'],
+    base_uri: %w['self'],
+    block_all_mixed_content: true, # see http://www.w3.org/TR/mixed-content/
+    child_src: %w['self'],
+    connect_src: %W['self'] + google_analytics,
+    font_src: %w['self' *.gov.uk fonts.gstatic.com],
+    form_action: %w['self'],
+    frame_ancestors: %w['self'],
+    frame_src: %w['self'],
+    img_src: %W['self' *.gov.uk] + google_analytics,
+    manifest_src: %w['self'],
+    media_src: %w['self'],
+    script_src: %W['self' 'unsafe-inline' 'unsafe-eval' *.gov.uk] + google_analytics,
+    style_src: %w['self' 'unsafe-inline' *.gov.uk] + google_analytics,
+    worker_src: %w['self'],
+    upgrade_insecure_requests: !Rails.env.development?, # see https://www.w3.org/TR/upgrade-insecure-requests/
+    report_uri: %w[/csp_reports],
```

Comment:
We should consider using the reporting endpoint from Sentry

Reply:
Added an internal route with logging to console with notifications. Or should I just log direct from controller